components/brave_wallet/browser/fil_transaction.cc

```diff
@@ -130,49 +138,94 @@ absl::optional<FilTransaction> FilTransaction::FromValue(
     tx.nonce_ = nonce;
   }
 
-  const std::string* gas_premium = value.FindStringKey("gas_premium");
+  const std::string* gas_premium = value.FindStringKey("GasPremium");
   if (!gas_premium)
     return absl::nullopt;
   tx.gas_premium_ = *gas_premium;
 
-  const std::string* gas_fee_cap = value.FindStringKey("gas_fee_cap");
+  const std::string* gas_fee_cap = value.FindStringKey("GasFeeCap");
   if (!gas_fee_cap)
     return absl::nullopt;
   tx.gas_fee_cap_ = *gas_fee_cap;
 
-  const std::string* max_fee = value.FindStringKey("max_fee");
+  const std::string* max_fee = value.FindStringKey("MaxFee");
   if (!max_fee)
     return absl::nullopt;
   tx.max_fee_ = *max_fee;
 
-  const std::string* gas_limit = value.FindStringKey("gas_limit");
+  const std::string* gas_limit = value.FindStringKey("GasLimit");
   if (!gas_limit || !base::StringToInt64(*gas_limit, &tx.gas_limit_))
     return absl::nullopt;
 
-  const std::string* to = value.FindStringKey("to");
+  const std::string* from = value.FindStringKey("From");
+  if (!from)
+    return absl::nullopt;
+  tx.from_ = FilAddress::FromAddress(*from);
+
+  const std::string* to = value.FindStringKey("To");
   if (!to)
     return absl::nullopt;
   tx.to_ = FilAddress::FromAddress(*to);
 
-  const std::string* tx_value = value.FindStringKey("value");
+  const std::string* tx_value = value.FindStringKey("Value");
   if (!tx_value)
     return absl::nullopt;
   tx.value_ = *tx_value;
-
   return tx;
 }
 
-std::string FilTransaction::GetMessageToSign() const {
+absl::optional<std::string> FilTransaction::GetMessageToSign() const {
+  auto value = ToValue();
+  value.RemoveKey("MaxFee");
+  value.SetIntKey("MethodNum", 0);
+  value.SetIntKey("Version", 0);
+  value.SetStringKey("Params", "");
+  std::string json;
+  base::JSONWriter::Write(value, &json);
+
+  std::string converted_json =
+      json::convert_string_value_to_int64("/GasLimit", json.c_str(), false)
+          .c_str();
+  if (converted_json.empty())
+    return absl::nullopt;
+  converted_json = json::convert_string_value_to_uint64(
+                       "/Nonce", converted_json.c_str(), false)
+                       .c_str();
+  return converted_json;
+}
+
+// https://spec.filecoin.io/algorithms/crypto/signatures/#section-algorithms.crypto.signatures
+absl::optional<std::string> FilTransaction::GetSignedTransaction(
+    const std::string& private_key_base64) const {
+  auto message = GetMessageToSign();
+  if (!message)
+    return absl::nullopt;
+  std::string data(
+      filecoin::transaction_sign(*message, private_key_base64).c_str());
```

Comment:
cxx [implements](https://github.com/dtolnay/cxx/blob/1.0.57/include/cxx.h#L52) `operator std::string` so I don't think you need the explicit `.c_str()` here.

Reply:
removed